Simulator/Simulator/Simulator/python/utils/utils_matplotlib.py

```diff
@@ -60,26 +195,93 @@ def line_plot(output_file, y_data, line_label, x_label, y_label, title):
         x_data = np.arange(y_data[i].shape[0])
         plt.plot(x_data, y_data[i], label=line_label[i])
 
-        x_data_min = x_data.min()
-        x_data_max = x_data.max()
-        y_data_min = y_data[i].min()
-        y_data_max = y_data[i].max()
+        if x_data.shape[0] != 0:
+            x_min, x_max = get_min_max(x_min, x_max, x_data)
+            y_min, y_max = get_min_max(y_min, y_max, y_data[i])
 
-        x_min = x_data_min if x_data_min < x_min else x_min
-        x_max = x_data_max if x_data_max > x_max else x_max
-        y_min = y_data_min if y_data_min < y_min else y_min
-        y_max = y_data_max if y_data_max > y_max else y_max
+    set_x_ticks(x_min, x_max)
+    set_y_ticks(y_min, y_max)
+    set_legend()
 
-    x_step = (x_max - x_min) / 40.0
-    if x_step != 0.0:
-        plt.xticks(np.arange(start=x_min, stop=x_max + x_step, step=x_step), rotation=90)
+    plt.grid(True, linestyle='--')
+    plt.title(title, fontsize=20)
+    plt.xlabel(x_label, fontsize=20)
+    plt.ylabel(y_label, fontsize=20)
+    plt.tight_layout()
 
-    y_step = (y_max - y_min) / 30.0
-    if y_step != 0.0:
-        plt.yticks(np.arange(start=y_min, stop=y_max + y_step, step=y_step))
+    plt.savefig(output_file, dpi=100)
+
+    reset_plot()
+
+
+def order_points_to_polygon(x_data, y_data):
+    '''
+    It orders a set of points so that they form a polygon
+    https://stackoverflow.com/questions/10846431/ordering-shuffled-points-that-can-be-joined-to-form-a-polygon-in-python
+    '''
+    pp = [(x, y) for x, y in zip(x_data, y_data)]
+    # sort by polar angle
+    cent = (sum([p[0] for p in pp]) / len(pp), sum([p[1] for p in pp]) / len(pp))
+    pp.sort(key=lambda p: math.atan2(p[1] - cent[1], p[0] - cent[0]))
+    return np.array([p[0] for p in pp]), np.array([p[1] for p in pp])
```

Comment:
I think this can be removed, but check plz

Reply:
Yes, it has been removed, because I give up on the idea of plotting in polygon style